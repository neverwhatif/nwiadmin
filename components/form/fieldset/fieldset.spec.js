import React from 'react';
import { shallow } from 'enzyme';

import Fieldset from './index';

jest.mock('nwiadmin/components/form/formlegend', () => 'FormLegend');

describe('Fieldset', () => {
    const component = shallow(<Fieldset><span>Lorem</span></Fieldset>);
    const componentWithLegend = shallow(<Fieldset legend="Lorem"><span>Lorem</span></Fieldset>);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentWithLegend).toMatchSnapshot();
    });
    it('should render a legend', () => {
        expect(component.find('FormLegend').length).toBe(0);
        expect(componentWithLegend.find('FormLegend').length).toBe(1);
        expect(componentWithLegend.find('FormLegend').prop('children')).toBe('Lorem');
    });
});
