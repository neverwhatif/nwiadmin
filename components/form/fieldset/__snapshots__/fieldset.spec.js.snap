// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Fieldset
    legend={null}
  >
    <span>
      Lorem
    </span>
  </Fieldset>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <span>
          Lorem
        </span>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "span",
      },
    ],
    "type": "fieldset",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <span>
            Lorem
          </span>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "span",
        },
      ],
      "type": "fieldset",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Fieldset should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Fieldset
    legend="Lorem"
  >
    <span>
      Lorem
    </span>
  </Fieldset>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FormLegend>
          Lorem
        </FormLegend>,
        <span>
          Lorem
        </span>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "FormLegend",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "span",
      },
    ],
    "type": "fieldset",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormLegend>
            Lorem
          </FormLegend>,
          <span>
            Lorem
          </span>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "FormLegend",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "span",
        },
      ],
      "type": "fieldset",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
